### ISSUE TITLE
Plugin fails with error Assignment to constant variable

### ISSUE THREAD

**Issue by gunters63** [naecoo/esbuild-plugin-replace#6]
I get sometimes this error message from the plugin:

![image](https://user-images.githubusercontent.com/90346/194243062-3fd14e2f-bdf6-4c76-a45d-c8c7aa21e2db.png)

I don't exactly why it works in some environments (Node, ESbuild version etc.).

But the problem is clear. The following code in index.js would normally lead to a runtime error:

````
const replaceCode = (code, id, pattern, functionValues) => {
  const magicString = new MagicString(code);
  while ((match = pattern.exec(code))) {
    const start = match.index;
    .....
````

because match is an undeclared Javascript variable.

The reason why it still works is the first line in the dist bundle:

![image](https://user-images.githubusercontent.com/90346/194243544-7d057d5d-a6f2-4f10-8aaa-e96217b2ec10.png)

match is already declared as a function. replaceCode redefines the function in the while loop, so it won't generate an undeclared Javascript variable error.

This normally works, but somehow in my environment it sometimes fails with `Assignment to constant variable` (I use Node 18.10)

Anyway, I think this is a bug and the fix should be easy. Just declare the match variable with `let match; ` inside replaceCode.

**Comment by naecoo** [comment]
Thanks, fixed in #7, and you can use version [1.2.1](https://github.com/naecoo/esbuild-plugin-replace/releases/tag/v1.2.1) now

**Comment by naecoo** [comment]
Thanks, fixed in https://github.com/naecoo/esbuild-plugin-replace/pull/7, and you can use version [1.2.1](https://github.com/naecoo/esbuild-plugin-replace/releases/tag/v1.2.1) now
